docs/features/multiple-networks/multi-homing.md

```diff
@@ -169,12 +169,66 @@ localnet network.
    IP addresses in a `ipamclaims.k8s.cni.cncf.io` object. This IP addresses will
    be reused by other pods if requested. Useful for KubeVirt VMs. Only makes
    sense if the `subnets` attribute is also defined.
+- `physicalNetworkName` (string, optional): the name of the physical network to
+  which the OVN overlay will connect. When omitted, it will default to the value
+  of the localnet network `name`.
```

Comment:
I am assuming this is specific to localnet. Should the doc be more clear that this is being ignored for other types?

What is the plan though for L2, L3 primaries that will also be connected to the outside world? Wouldn't we want parity?

Reply:
> I am assuming this is specific to localnet. Should the doc be more clear that this is being ignored for other types?

It is specific to localnet - it is *exclusively* defined in the localnet network configuration reference [section](https://github.com/ovn-org/ovn-kubernetes/blob/1c92c4d39cb1fb2b9cb22c292f047db52e570f2e/docs/features/multiple-networks/multi-homing.md#network-configuration-reference-2). 

I don't know how to make it clearer than that. I can write there that it only applies to localnet, but again, this section **is** for localnet only. 

The same comment would apply to `vlanID`.

> What is the plan though for L2, L3 primaries that will also be connected to the outside world? Wouldn't we want parity?

I think L2 / L3 primaries have a lot more batteries included and are a lot more "opinionated" than secondary networks. 

We *may* want to have parity in the future (i.e. a configurable mapping to the physical network for the primary network), but right now, nobody's asking for it. I do think it is a reasonable ask though.

Reply:
Do the L2 / L3 primaries ingress/egress through the same localnet port as the default network?

Reply:
> Do the L2 / L3 primaries ingress/egress through the same localnet port as the default network?

No.

Reply:
I am just trying to guess which bridge mapping they are using. Do they hardcode `physnet` as network name then?

All this was more clear when there were just secondary networks and localnet was the only topology connected with the outside world. We could say this document is specifically for secondary networks. An explicit acknowledgement that it does not cover primary networks could also help. It's confusing because there is no equivalent doc for primaries, no doc covering the `Role` attribute of the NAD, and the last section on this doc covering a topic on primaries further confuses things. And to add to the confusion, you go to the code and see that `IsSecondary` returns true for primaries as well.

So if I see `physicalNetworkName` here, even if it is not documented in other sections, I would ask myself `Why wouldn't it work for other topologies for which this would make sense?` And then I would go around looking for other hints and probably be misguided. 

I am just saying this as an outsider that has not been deeply involved in UDN. I had that question and a clarification that explicitly states `This only applies to localnet secondary networks` would have helped me at a user level. But if you think it is fine then lets go with it.

Reply:
I acknowledge all those limitations - but, to me, what's really missing is a document explaining primary UDN features.

FWIW, I think we should also do a major overhaul of the multi-homing docs after primary UDN lands. 

@RamLavi started something like that, but with a different scope in https://github.com/ovn-org/ovn-kubernetes/pull/4537

Reply:
Do we have some way to track that future effort?

Reply:
Do you mean the primary UDN docs ? If so, yes: https://issues.redhat.com/browse/SDN-5425